workspaces/spectacle/src/index.ts

```diff
@@ -311,6 +312,35 @@ export async function makeSpectacle(opticContext: IOpticContext) {
           .spectacleContext()
           .opticContext.diffRepository.findById(diffId);
       },
+      metadata: async (parent: any, args: any, context: any, info: any) => {
+        const metadataKey = 'metadata';
+        const specIdKey = 'id';
+        const metadata =
+          context.spectacleContext().contributionsProjection[metadataKey] || {};
+        let specId = metadata[specIdKey];
+
+        if (!specId) {
+          specId = uuidv4();
+          await context
```

Comment:
So this causes spectacle to be reset (which is intended), but this means on:
- Public examples
- Cloud viewer with no specId metadata
- LocalCLI with no specId

The spectacle instance will fetch twice on load (i.e. initial load will set spectacle, and then this will cause spectacle to reload)

This is part of the reason I blocked loading so only the spec metadata load is sent at the start